### PR TITLE
Add some descriptions

### PR DESCRIPTION
**Pull request by SpacePossum**

### CODE REVIEW

File: tests/FixerFactoryTest.php [hunk 1]
```diff
@@ -454,7 +455,7 @@ public function provideFixersForFinalCheckCases()
      */
     public function testShortFixerDefinition()
     {
-        $guard = 118;
+        $guard = 102;
```
Comment (keradus) [inline tests/FixerFactoryTest.php]: wow, nice !

File: src/Tokenizer/Tokens.php [hunk 2]
```diff
@@ -1083,6 +1083,13 @@ public function isMonolithicPhp()
             ) {
                 return false;
             }
+
+            if ($this[$index]->isGivenKind(T_CLOSE_TAG)) {
```
Comment (SpacePossum) [inline src/Tokenizer/Tokens.php]: @keradus besides all the descriptions, I would like you to take a look at this fix :)

File: tests/FixerFactoryTest.php (hunk 1 above)
Comment (SpacePossum) [inline tests/FixerFactoryTest.php]: 83 and counting ;)

File: src/Tokenizer/Tokens.php (hunk 2 above)
Comment (keradus) [inline src/Tokenizer/Tokens.php]: I think that this should be separated PR.
One PR with descriptions, one with logic change

**Comment by SpacePossum** [comment]
all green? something must be wrong with the tests : p

**Comment by SpacePossum** [comment]
If this is green lets get it merged before it comes a conflict-hell ^_^

**Comment by GrahamCampbell** [comment]
👍

File: README.rst [hunk 3]
```diff
@@ -296,8 +297,8 @@ Choose from the list of available rules:
    | Methods must be separated with one blank line.
 
 * **modernize_types_casting**
-   | Replaces intval, floatval, doubleval, strval, boolval functions calls
-   | with according type casting operator.
+   | Replaces `intval`, `floatval`, `doubleval`, `strval` and `boolval
+   | `functions calls with according type casting operator.
```
Comment (localheinz) [inline README.rst]: Looks like something is up with the wrapping here, no?

File: src/Fixer/CastNotation/ModernizeTypesCastingFixer.php [hunk 4]
```diff
@@ -113,8 +107,39 @@ public function fix(\SplFileInfo $file, Tokens $tokens)
     /**
      * {@inheritdoc}
      */
-    protected function getDescription()
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Replaces `intval`, `floatval`, `doubleval`, `strval` and `boolval `functions calls with according type casting operator.',
```
Comment (localheinz) [inline src/Fixer/CastNotation/ModernizeTypesCastingFixer.php]: Should this be

```diff
-'Replaces `intval`, `floatval`, `doubleval`, `strval` and `boolval `functions calls with according type casting operator.',
+'Replaces `intval`, `floatval`, `doubleval`, `strval` and `boolval` function calls with according type casting operator.',
```

Comment (SpacePossum) [inline src/Fixer/CastNotation/ModernizeTypesCastingFixer.php]: indeed, impressive spotting in all those changes :)   will update!

File: README.rst (hunk 3 above)
Comment (keradus) [inline README.rst]: indeed !

File: tests/Fixer/Semicolon/NoSinglelineWhitespaceBeforeSemicolonsFixerTest.php [hunk 5]
```diff
@@ -90,9 +90,6 @@ public function provideCases()
     ;',
             ),
             array(
-                '<?php $this->foo();',
```
Comment (keradus) [inline tests/Fixer/Semicolon/NoSinglelineWhitespaceBeforeSemicolonsFixerTest.php]: why ?

File: tests/RuleSetTest.php [hunk 6]
```diff
@@ -228,6 +228,43 @@ public function providerSetDefinitionNames()
...
+     */
+    public function testOnlyRiskyRulesInSet(array $set)
+    {
+        $this->assertSetSafety($set, false);
+    }
+
+    public function provideRiskySets()
+    {
+        return array(array(array(
+            '@Symfony:risky' => true,
```
Comment (keradus) [inline tests/RuleSetTest.php]: please add `['@Symfony:risky' => true,]` case

File: tests/RuleSetTest.php [hunk 7]
```diff
@@ -228,6 +228,43 @@ public function providerSetDefinitionNames()
         }, $setDefinitionNames);
     }
 
+    /**
+     * @param array $set
+     *
+     * @dataProvider provideSafeSets
+     */
+    public function testNoRiskyRulesInSet(array $set)
```
Comment (keradus) [inline tests/RuleSetTest.php]: combine with `testOnlyRiskyRulesInSet` and use 2nd argument that will be expected safety

File: tests/Fixer/Semicolon/NoSinglelineWhitespaceBeforeSemicolonsFixerTest.php [hunk 8]
```diff
@@ -10,7 +10,7 @@
  * with this source code in the file LICENSE.
  */
 
-namespace PhpCsFixer\Tests\Fixer\Semicolon;
+namespace PhpCsFixer\tests\Fixer\Semicolon;
```
Comment (SpacePossum) [inline tests/Fixer/Semicolon/NoSinglelineWhitespaceBeforeSemicolonsFixerTest.php]: indeed .. what?

File: tests/RuleSetTest.php [hunk 9]
```diff
@@ -237,6 +274,38 @@ private function assertSameRules(array $expected, array $actual, $message = '')
     }
 
     /**
+     * @param array $set
+     * @param bool  $safe
+     */
+    private function assertSetSafety(array $set, $safe)
+    {
+        $set = RuleSet::create($set);
```
Comment (keradus) [inline tests/RuleSetTest.php]: `new RuleSet($set)`
or
`RuleSet::create($set)->getRules()`

File: tests/Fixer/Semicolon/NoSinglelineWhitespaceBeforeSemicolonsFixerTest.php (hunk 8 above)
Comment (keradus) [inline tests/Fixer/Semicolon/NoSinglelineWhitespaceBeforeSemicolonsFixerTest.php]: it should not be lowercased

Comment (SpacePossum) [inline tests/Fixer/Semicolon/NoSinglelineWhitespaceBeforeSemicolonsFixerTest.php]: I know, wonder what happend

Comment (keradus) [inline tests/Fixer/Semicolon/NoSinglelineWhitespaceBeforeSemicolonsFixerTest.php]: did you apply wrongly fabbot.io patch trying to enforce psr-0 perhaps ?

File: tests/RuleSetTest.php [hunk 10]
```diff
@@ -279,14 +275,11 @@ private function assertSameRules(array $expected, array $actual, $message = '')
...
-        $rules = $set->getRules();
-        $this->assertNotEmpty($rules);
-
-        $factory = FixerFactory::create();
-        $factory->registerBuiltInFixers();
-        $factory->useRuleSet($set);
-        $fixers = $factory->getFixers();
+        $fixers = FixerFactory::create()
+            ->registerBuiltInFixers()
+            ->useRuleSet(RuleSet::create($set))
```
Comment (keradus) [inline tests/RuleSetTest.php]: please use `new` keyword when possible (when not chaining)

File: tests/FixerFactoryTest.php [hunk 11]
```diff
@@ -530,6 +531,60 @@ public function testNoDoubleConflictReporting()
...
+            }
+        }
+
+        if (count($problems)) {
+            $list = '';
+            foreach ($problems as $fixerName => $methodNames) {
+                $list .= sprintf("\nFixer: %s\n%s\n", $fixerName, implode("\n", $methodNames));
+            }
+
+            $this->fail(sprintf("Fixers should not have public methods that are not part of the interfaces.\nViolations:%s", $list));
```
Comment (keradus) [inline tests/FixerFactoryTest.php]: great check !
let us extract it for separated PR to have proper changelog

File: tests/FixerFactoryTest.php [hunk 12]
```diff
@@ -530,6 +531,60 @@ public function testNoDoubleConflictReporting()
         );
     }
 
+    public function testFixerEndPoints()
```
Comment (keradus) [inline tests/FixerFactoryTest.php]: test should be atomic, check one fixer and use data provider to provide all fixers step by step

**Comment by SpacePossum** [comment]
closing in fav. of https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/2400

**Pull request closed without merging**